packages/editor/src/components/post-author/combobox.js

```diff
@@ -57,22 +61,17 @@ function PostAuthorCombobox() {
 
 		if ( foundAuthor < 0 && postAuthor ) {
 			return [
-				{ value: postAuthor.id, label: postAuthor.name },
+				{
+					value: postAuthor.id,
+					label: decodeEntities( postAuthor.name ),
+				},
 				...fetchedAuthors,
 			];
 		}
 
 		return fetchedAuthors;
 	}, [ authors, postAuthor ] );
 
-	// Initializes the post author properly
-	// Also ensures external changes are reflected.
-	useEffect( () => {
-		if ( postAuthor ) {
-			setFieldValue( postAuthor.name );
-		}
-	}, [ postAuthor ] );
-
```

Comment:
I couldn't track down why this effect was added precisely, but I don't think it's needed anymore.

It also initializes an unnecessary search query on the mount.